src/INode.java

```diff
@@ -320,9 +323,17 @@ private ASTValue evalParameterList(WordsEnvironment environment) {
 		throw new AssertionError("Not yet implemented");
 	}
 
-	private ASTValue evalPosition(WordsEnvironment environment) {
-		ASTValue row = children.get(0).eval(environment);
-		ASTValue col = children.get(1).eval(environment);
+	private ASTValue evalPosition(WordsEnvironment environment) throws WordsProgramException {
+		ASTValue row = children.get(0).eval(environment).getNumCoercedVal();
+		ASTValue col = children.get(1).eval(environment).getNumCoercedVal();
+		
+		if (row.type != ValueType.NUM) {
```

Comment:
You have a potential null pointer exception here (and in other similar places), because getNumCoercedVal() can return null.